### PR TITLE
[sharp] remove package in favor of official types

### PR DESCRIPTION
**Pull request by rexxars**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If removing a declaration:
- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

Removes the `sharp` typings since they are now in the official package as of 0.32.0.

### CODE REVIEW

File: types/11ty__eleventy-img/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "dependencies": {
+        "@types/sharp": "^0.31.0"
```
Comment (NotWoods) [inline types/11ty__eleventy-img/package.json:4]: These should get updated to reference the new bundled types
```suggestion
        "sharp": "^0.32.0"
```

**Review by NotWoods — CHANGES_REQUESTED** [review]

File: types/11ty__eleventy-img/package.json (hunk 1 above)
Comment (rexxars) [reply types/11ty__eleventy-img/package.json:4]: I would say that belongs in a separate PR. It is only in `@11ty/eleventy-img` v3.1.0 which was released 4 days ago that it uses 0.32 - prior to that it was using the 0.31 range. The `@types/11ty__eleventy-img` module should mirror that and release a 3.1.0 which includes the 0.32 typings.

**Review by NotWoods — APPROVED** [review]

**Review by Dattaya — APPROVED** [review]

**Review by NotWoods — APPROVED** [review]

**Review by Dattaya — APPROVED** [review]

**Review by sheetalkamat — DISMISSED** [review]
All the typings need to update to bundled types per https://github.com/DefinitelyTyped/DefinitelyTyped#removing-a-package

**Comment by rexxars** [comment]
> All the typings need to update to bundled types per https://github.com/DefinitelyTyped/DefinitelyTyped#removing-a-package

Hi @sheetalkamat - could you explain what this means in this context?  I've already changed the type modules that were depending on `@types/sharp` to add it as a dependency:

- [staticmaps](https://github.com/StephanGeorg/staticmaps/blob/master/package.json#L42) depends on `0.31.3`, so should not use 0.32 typings, as that would bring in two versions of the module, and according to [semver](https://docs.npmjs.com/cli/v6/using-npm/semver#caret-ranges-123-025-004) is not semver compatible.
- [sharpie](https://github.com/kapouer/sharpie/blob/4.5.3/package.json#L33) 4.5.x (which the types module mirrors) actually depends on `sharp@0.24.1`. Theoretically it should be using `@types/sharp@^0.24.0` to mirror the version it actually uses internally.
- [lqip-modern](https://github.com/transitive-bullshit/lqip-modern/blob/v1.1.3/package.json#L23) 1.1.x (which the types module mirrors) actually depends on `sharp@^0.25.4`. Theoretically it should be using `@types/sharp@^0.25.0` to mirror the version it actually uses internally.
- @11ty/eleventy-img only switched to sharp 0.32 in their recent [3.1.0](https://github.com/11ty/eleventy-img/releases/tag/v3.1.0) release, and the latest typings for this module is still on the [2.x range of the module](https://www.npmjs.com/package/@types/11ty__eleventy-img). As I mentioned in an earlier comment, that type module needs a separate PR that both provides typings for the 3.x version, and can then use the 0.32.0 module. The current version of the type module should stick with 0.31-compatible typings, in my opinion.

Since all these modules were implicitly depending on the latest (0.31) range at the time of this PR without any reports of issues, I have used that version for these, but as I've stated - if anything these should be depending on _older_ versions of the type module, not newer.

Have I misunderstood how this is supposed to work? All tests are passing.

**Comment by rexxars** [comment]
@sheetalkamat Bump - any chance you can explain/re-review based on my previous comment?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@rexxars sorry for a delay, there is a rota in MS folks answering here directly,
thanks for your patience

**Comment by rexxars** [comment]
Ready to merge

**Pull request merged**